libstb/secvar/backend/edk2-compat-util.c

```diff
@@ -0,0 +1,601 @@
+ * name || vendor guid || attributes || timestamp || newcontent 
+ * which is submitted as signed by the user.
+ * Returns number of bytes in the new buffer, else negative error
+ * code.
+ */
+int get_data_to_verify(char *key, char *new_data,
+		size_t new_data_size,
+		char **buffer,
+		size_t *buffer_size, struct efi_time *timestamp)
+{
+	le32 attr = cpu_to_le32(SECVAR_ATTRIBUTES);
+	size_t offset = 0;
+	size_t varlen;
+	char *wkey;
+	uuid_t guid;
+
+	if (key_equals(key, "PK")
+	    || key_equals(key, "KEK"))
+		guid = EFI_GLOBAL_VARIABLE_GUID;
+	else if (key_equals(key, "db")
+	    || key_equals(key, "dbx"))
+		guid = EFI_IMAGE_SECURITY_DATABASE_GUID;
+		
+	// Convert utf8 name to ucs2 width
+	varlen = strlen(key) * 2;
+	wkey = utf8_to_ucs2(key, strlen(key));
+
+	// Prepare the single buffer
+	*buffer_size = varlen + UUID_SIZE + sizeof(attr)
+		       + sizeof(struct efi_time) + new_data_size;
+	*buffer = zalloc(*buffer_size);
+	if (!*buffer)
+		return OPAL_NO_MEM;
+
+	memcpy(*buffer + offset, wkey, varlen);
+	offset = offset + varlen;
+	memcpy(*buffer + offset, &guid, sizeof(guid));
+	offset = offset + sizeof(guid);
+	memcpy(*buffer + offset, &attr, sizeof(attr));
+	offset = offset + sizeof(attr);
+	memcpy(*buffer + offset, timestamp , sizeof(struct efi_time));
+	offset = offset + sizeof(struct efi_time);
+
+	memcpy(*buffer + offset, new_data, new_data_size);
+	offset = offset + new_data_size;
+
+	free(wkey);
+
+	return offset;
+}
+
+bool is_pkcs7_sig_format(void *data)
+{
+	struct efi_variable_authentication_2 *auth = data;
+	uuid_t pkcs7_guid = EFI_CERT_TYPE_PKCS7_GUID;
+
+	if(!(memcmp(&auth->auth_info.cert_type, &pkcs7_guid, 16) == 0))
+		return false;
+
+	return true;
+}
+
+int process_update(struct secvar_node *update, char **newesl,
+		   int *new_data_size, struct efi_time *timestamp)
+{
+        char *auth_buffer = NULL;
+        int auth_buffer_size = 0;
+        const char *key_authority[3];
+        char *tbhbuffer = NULL;
+        size_t tbhbuffersize = 0;
+        struct secvar_node *anode = NULL;
+        int rc = 0;
+        int i;
+
+	auth_buffer_size = get_auth_descriptor2(update->var->data,
+						update->var->data_size,
+						&auth_buffer);
+	if ((auth_buffer_size < 0)
+	     || (update->var->data_size < auth_buffer_size)) {
+		rc = auth_buffer_size;
+		goto out;
+	}
+
+	if (!timestamp) {
+		rc = OPAL_INTERNAL_ERROR;
+		goto out;
+	}
+	
+	memcpy(timestamp, auth_buffer, sizeof(struct efi_time));
+
+	rc = check_timestamp(update->var->key, timestamp);
+	/* Failure implies probably an older command being resubmitted */
+	if (rc != OPAL_SUCCESS)
+		goto out;
+
+	/* Calculate the size of new ESL data */
+	*new_data_size = update->var->data_size - auth_buffer_size;
+	if (*new_data_size < 0) {
+		rc = OPAL_PARAMETER;
+		goto out;
+	}
+	*newesl = zalloc(*new_data_size);
+	if (!(*newesl)) {
+		rc = OPAL_NO_MEM;
+		goto out;
+	}
+	memcpy(*newesl, update->var->data + auth_buffer_size, *new_data_size);
+
+	if (!setup_mode) {
```

Comment:
Reverse the condition here, and jump to `out`. That way the following code saves a level of indentation